### DISCUSSION TITLE
PDF goes blank for half a second after each highlight

### DISCUSSION THREAD

**Discussion by JacobHamelMottiez** [RyotaUshio/obsidian-pdf-plus#152] [Q&A] [answered]
Hi,

Thanks for this amazing plugin! I was wondering why after each highlight the PDF goes blank and then reappear with the highlighted text. Is it a normal behaviour? 

Best, 
Jacob

**Comment by RyotaUshio** [comment]
That sounds strange. 

- Can you share the PDF and a screen recording?
- Does it happen to all PDFs? Or only to specific PDFs? If so, what kind of PDFs (e.g. large size, many pages etc)?

**Reply by JacobHamelMottiez** [reply]
Hi @RyotaUshio, 

Thanks for your answer! Yes it does it with all the PDFS. I created an empty vault with only the PDF++ plugin to rule out possible interference with other plugins. 

Best, 

https://github.com/RyotaUshio/obsidian-pdf-plus/assets/122307017/52d5d259-7909-4dbc-b2f6-509aa03607e7

**Reply by RyotaUshio** [reply, marked as answer]
Oh now I see what's happening. This is in fact an expected behavior. It happens because you are adding highlights directly to the file.

Even without PDF++, Obsidian's native PDF viewer **fully refreshes** when the PDF file is modified. Since you are modifying the PDF file itself (<img width="30" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/ecda2eba-87f5-4c03-87f2-c31fd4b0c05a">), it causes the viewer to reload every time you add a new highlight. This is how the Obsidian team implemented the viewer (rather than me), and it's not very easy to change this behavior.
But of course, I do want to figure out a trick to make it behave more nicely someday. It's just not what I can do in the short term.

If this flickering really annoys you, unfortunately, you need to add highlights without modifying the file itself (i.e. with [backlink highlighting](https://ryotaushio.github.io/obsidian-pdf-plus/backlink-highlighting-basics.html)). You can do so by disabling the <img width="30" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/ecda2eba-87f5-4c03-87f2-c31fd4b0c05a"> icon.
Alternatively, you can use another PDF editing app like Adobe Acrobat to add highlights - PDF++ has some features that make it easier to integrate such apps with Obsidian (see the "integration with external apps" section near the bottom of the plugin setting tab).

**Reply by JacobHamelMottiez** [reply]
I see, It clarifies things!  Thanks again for this amazing plugin, I hope that this annotation behavior will be "nicer" someday!

Best, 
Jacob

**Comment by N3C2L** [comment]
@RyotaUshio 
Do you think you can delay this writings to the PDF file until a user clicks on (a new button) "save temporary highlights to the PDF file" ?
Maybe the informations about this temporary highlights could be stored or clipboarded anywhere else until a user clicks this new button? 
Because in other PDF software, the user also has to click "save" to write the highlights to the PDF directly.

**Comment by RyotaUshio** [comment]
Update: You might want to try this new plugin: https://github.com/Quorafind/Obsidian-PDF-Annotator

My note here: https://github.com/RyotaUshio/obsidian-pdf-plus/blob/main/README.md#note-on-saving-annotations-directly-in-pdf
